js/btcturk.js

```diff
@@ -518,53 +538,132 @@ module.exports = class btcturk extends Exchange {
     }
 
     parseOHLCV (ohlcv, market = undefined) {
-        //     {
-        //        "pair": "BTCTRY",
-        //        "time": 1508284800,
-        //        "open": 20873.689453125,
-        //        "high": 20925.0,
-        //        "low": 19310.0,
-        //        "close": 20679.55078125,
-        //        "volume": 402.216101626982,
-        //        "total": 8103096.44443274,
-        //        "average": 20146.13,
-        //        "dailyChangeAmount": -194.14,
-        //        "dailyChangePercentage": -0.93
-        //      },
+        //
+        //    {
+        //        't': 1661990400,
+        //        'h': 368388.0,
+        //        'o': 368388.0,
+        //        'l': 368388.0,
+        //        'c': 368388.0,
+        //        'v': 0.00035208,
+        //    }
+        //
         return [
-            this.safeTimestamp (ohlcv, 'time'),
-            this.safeNumber (ohlcv, 'open'),
-            this.safeNumber (ohlcv, 'high'),
-            this.safeNumber (ohlcv, 'low'),
-            this.safeNumber (ohlcv, 'close'),
-            this.safeNumber (ohlcv, 'volume'),
+            this.safeTimestamp (ohlcv, 't'),
+            this.safeNumber (ohlcv, 'o'),
+            this.safeNumber (ohlcv, 'h'),
+            this.safeNumber (ohlcv, 'l'),
+            this.safeNumber (ohlcv, 'c'),
+            this.safeNumber (ohlcv, 'v'),
         ];
     }
 
-    async fetchOHLCV (symbol, timeframe = '1d', since = undefined, limit = undefined, params = {}) {
+    async fetchOHLCV (symbol, timeframe = '1h', since = undefined, limit = undefined, params = {}) {
         /**
          * @method
          * @name btcturk#fetchOHLCV
          * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://docs.btcturk.com/public-endpoints/get-kline-data
          * @param {string} symbol unified symbol of the market to fetch OHLCV data for
          * @param {string} timeframe the length of time each candle represents
          * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
          * @param {int|undefined} limit the maximum amount of candles to fetch
          * @param {object} params extra parameters specific to the btcturk api endpoint
+         * @param {int|undefined} params.until timestamp in ms of the latest candle to fetch
          * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
-            'pair': market['id'],
+            'symbol': market['id'],
+            'resolution': this.timeframes[timeframe],
         };
+        const until = this.safeInteger (params, 'until', this.milliseconds ());
+        if (since !== undefined) {
+            request['from'] = parseInt (since / 1000);
+        }
+        if (until !== undefined) {
```

Comment:
you're using `this.milliseconds()` as the default value, so `until` will never be undefined